### ISSUE TITLE
Java highlight groups with `.` lead to "Invalid character in group name" messages

### ISSUE THREAD

**Issue by mfussenegger** [nvim-treesitter/nvim-treesitter#1621]
**Describe the bug**

Using vim treesitter highlighting on a Java file that contains certain expressions lead to a `W18: Invalid character in group name` message.

**To Reproduce**

1. Ensure the java highlights.scm file from the nvim-treesitter is in the runtime path
2. Create a java file with the following contents:

```java
public class Foo {

  public int foo() {
      return 1 + 1;
  }
}
```

3. Enable syntax highlighting:

```vimL
:lua vim.treesitter.highlighter.new(vim.treesitter.get_parser(0, vim.bo.filetype), require('vim.treesitter.query').get_query(vim.bo.filetype, 'highlights'))
```

You should see the warning message appear.

I debugged a bit and saw that the following highlight group is set using `nvim_exec`:

```vimL
highlight default link javakeyword.return keyword.return
```

Having a `.` in highlight groups is not supported and causes the warning.
I think other then the return expression there are some more expressions that can lead to highlight groups containing dots.

**Expected behavior**

No warning

**Output of `nvim --version`**

Happens with both nvim-0.5 and latest nightly (`NVIM v0.6.0-dev+64-g754ac5a83`)

**Comment by theHamsta** [comment]
I guess this would be the fix https://github.com/neovim/neovim/pull/14767 Neovim does not have a mapping for the highlight capture `@keyword.return` so it shouldn't blindly transform it to a highlight group

**Comment by theHamsta** [comment]
Normally, nvim-treesitter would set a mapping in `hl_map` but a missing entry shouldn't lead to an error

**Comment by theHamsta** [comment]
No, it's really an issue that Neovim thinks `keyword.return` is a highlight group. The PR should fix it to just not highlight it when there is no mapping.
